cpp/src/arrow/compute/kernels/scalar_temporal_test.cc

```diff
@@ -1070,6 +1070,9 @@ TEST_F(ScalarTemporalTest, TestTemporalAddDateAndDuration) {
         ArrayFromJSON(timestamp(TimeUnit::MICRO), times_seconds_precision);
     CheckScalarBinary(op, dates32, durations_us, timestamps_us);
     CheckScalarBinary(op, dates64, durations_us, timestamps_us);
+
+    CheckScalarBinary(op, durations_us, dates32, timestamps_us);
+    CheckScalarBinary(op, durations_us, dates64, timestamps_us);
```

Comment:
Here's a thought: how about we define `CheckScalarBinaryCommutative` so that we can simply replace `CheckScalarBinary` with it in all heterogenous addition tests?

Reply:
Nice! Done.